core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -130,6 +130,11 @@ class AclAuthorizer extends Authorizer with Logging {
 
   @volatile
   private var aclCache = new scala.collection.immutable.TreeMap[ResourcePattern, VersionedAcls]()(new ResourceOrdering)
+
+  @volatile
+  private var resourceCache = new scala.collection.immutable.HashMap[AccessControlEntry,
+    scala.collection.mutable.HashSet[ResourcePattern]]()
```

Comment:
commit 230ee36b9147a11d7ce299aa9fcbb590324faf68
Added the authorizeByResourceType() API to the benchmark and simulate the worst case: every allow ACL on the same resource has a dominant deny ACL.
Adjust the `resourceCount` parameter to "10000", "40000", "80000" since each cluster is unlikely to have more than 10k resources. Also, since we are testing against the worst case mentioned above, I think the "10000" cases are adequate for us.
Performance result here: https://paste.ubuntu.com/p/k8kV3w6fvF/